tests/agent/tools/test_session_workspace_isolation.py

```
@@ -0,0 +1,384 @@
+    root, session_dir, write, _ = _mk_env(tmp_path)
+    (root / "memory").mkdir()
+    target = root / "memory" / "MEMORY.md"
+    result = await write.execute(path=str(target), content="mem")
+    assert "Successfully wrote" in result
+    assert target.read_text(encoding="utf-8") == "mem"
+    assert not (session_dir / "memory" / "MEMORY.md").exists()
+
+
+@pytest.mark.asyncio
+async def test_write_file_outside_workspace_untouched(tmp_path):
+    root, _, write, _ = _mk_env(tmp_path)
```

Ruff (0.16.1)

[warning] Unpacked variable `root` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)